### DISCUSSION TITLE
K8S Deployments Mysql on juicefs

### DISCUSSION THREAD

**Discussion by dzbaiyang** [juicedata/juicefs#3429] [Q&A] [answered]
Today I want to try to deploy a mysql high availability on top of Juicefs. The configuration is as follows:

- ### **`juicefs-secret`**

apiVersion: v1
kind: Secret
metadata:
  name: juicefs-secret
  namespace: mysql
type: Opaque
stringData:
  name: juicefs-mysql
  metaurl: mysql://juicefs:hirisun@(192.168.13.132:3306)/jfs
  storage: minio
  bucket: http://192.168.13.133:9001/jfs
  access-key: qxoie8844QgUqHcR
  secret-key: jvJkXF2XIGQFtU7KI6Z2YbzatAwjOuXb
  envs: "{TZ: Asia/Shanghai}"

- ### **`StorageClass`**

apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: juicefs-sc
provisioner: csi.juicefs.com
parameters:
  csi.storage.k8s.io/provisioner-secret-name: juicefs-secret
  csi.storage.k8s.io/provisioner-secret-namespace: mysql
  csi.storage.k8s.io/node-publish-secret-name: juicefs-secret
  csi.storage.k8s.io/node-publish-secret-namespace: mysql

- ### **`pvc`**

apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: juicefs-mysql-pvc
  namespace: mysql
spec:
  accessModes:
  - ReadWriteMany
  resources:
    requests:
      storage: 10Pi
  storageClassName: juicefs-sc

- ### **`mysq-deploymentl`**

apiVersion: v1
kind: Deployment
metadata:
  labels:
    app: mysql
  name: mysql
  namespace: mysql
spec:
  selector:
    matchLabels:
      app: mysql
  template:
    metadata:
      labels:
        app: mysql
    spec:
      containers:
      - image: mysql:8.0
        name: mysql
        env:
        - name: MYSQL_ROOT_PASSWORD
          valueFrom:
            secretKeyRef:
              name: mysql-root-password
              key: password
        ports:
        - containerPort: 3306
        volumeMounts:
        - mountPath: /var/lib/mysql
          name: mysqlvolume
      volumes:
        - name: mysqlvolume
          persistentVolumeClaim:
            claimName: juicefs-mysql-pvc

- ### **`mysq-service`**

apiVersion: v1
kind: Service
metadata:
  labels:
    app: svc-mysql
  name: svc-mysql
  namespace: mysql
spec:
  selector:
    app: mysql
  type: NodePort
  ports:
  - port: 3306
    protocol: TCP
    targetPort: 3306
    nodePort: 31234

- ### **error**
rpc error: code = Internal desc = Could not delVol in juicefs: waitUtilJobCompleted: job juicefs-f02389eb-delvol isn't completed: 2023/04/04 00:38:14.057099 juicefs[7] <INFO>: Meta address: mysql://juicefs:****@(192.168.13.132:3306)/jfs [interface.go:402] 2023/04/04 00:38:14.063897 juicefs[7] <INFO>: Data use minio://192.168.13.133:9001/jfs/juicefs-mysql/ [format.go:429] 2023/04/04 00:38:14.120015 juicefs[7] <FATAL>: format: cannot update volume name from juicefs to juicefs-mysql [format.go:464]

MountVolume.SetUp failed for volume "pvc-2acee67e-8d4b-432f-8c3c-309b53781226" : rpc error: code = Internal desc = Could not mount juicefs: mount mysql://juicefs:****@(192.168.13.132:3306)/jfs at /jfs/pvc-2acee67e-8d4b-432f-8c3c-309b53781226-bptccq failed: 2023/04/04 00:35:10.969042 juicefs[7] <INFO>: Meta address: mysql://juicefs:****@(192.168.13.132:3306)/jfs [interface.go:402] 2023/04/04 00:35:10.972988 juicefs[7] <INFO>: Data use minio://192.168.13.133:9001/jfs/juicefs-mysql/ [format.go:429] 2023/04/04 00:35:11.009872 juicefs[7] <FATAL>: format: cannot update volume name from juicefs to juicefs-mysql [format.go:464]

**Comment by xiaogaozi** [comment, marked as answer]
> format: cannot update volume name from juicefs to juicefs-mysql 

This means that there is already a formatted JuiceFS file system in the MySQL database, and you need to replace it with a new MySQL database.

**Reply by dzbaiyang** [reply]
you are right , I had create new mysql database it's worked.
